tests/test_sdk.py

```diff
@@ -279,6 +284,89 @@ def test_job(hd_client):
             assert param[0] in fake_out.getvalue()
             assert str(param[1]) in fake_out.getvalue()
 
+    def test_experiment(self):
```

Comment:
There is one failure scenario I'm worried about. What happens if the user's code experiences an exception before end() is called? Can you try that manually? Not sure if we need a test for it yet, but we might...

Reply:
Yeah you found issues with error handling and the wrapper before. Makes sense to test for it.